api/src/test/java/org/apache/iceberg/expressions/TestTimestampLiteralConversions.java

```diff
@@ -107,6 +108,28 @@ public void testTimestampMicrosToDateConversion() {
     assertThat(dateOrdinal).isEqualTo(-1);
   }
 
+  @Test
+  public void testTimestampNanoWithLongLiteral() {
```

Comment:
could we also add a case for `withZone`?

Reply:
Sure, added the test case.